lib/src/rules/use_getters_to_access_properties.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.use_getters_to_access_properties;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/standard_resolution_map.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc =
+    r'Use getters for operations that conceptually access properties.';
+
+const _details = r'''
+
+**DO** use getters for operations that conceptually access properties.
+
+**BAD:**
+```
+rectangle.getWidth()
+collection.isEmpty()
+button.canShow()
+```
+
+**GOOD:**
+```
+rectangle.width
+collection.isEmpty
+button.canShow
+```
+
+''';
+
+bool _hasSideEffect(AstNode node) {
+  return (node is MethodInvocation ||
+      node is FunctionExpressionInvocation ||
+      node is AssignmentExpression ||
+      node is PrefixExpression ||
+      node is PostfixExpression);
```

Comment:
We might want to include a few other cases in this list, even though they are admittedly unlikely cases:
 - Factory constructors can do any amount of work, including causing side-effects in existing objects.
 - User definable operators can have side effects (they're just a method invocation). That includes the index operator `[]`.
- Anything that will implicitly invoke `toString()` because it could have a side-effect (such as string interpolation).
I leave it to you to decide whether these are worth guarding against.